deployers/whiskclient.go

```diff
@@ -20,194 +20,220 @@ package deployers
+		if utils.FileExists(manifestPath) {
+			mm := parsers.NewYAMLParser()
+			manifest, _ := mm.ParseManifest(manifestPath)
+			credential = GetPropertyValue(credential, manifest.Package.Credential, path.Base(manifestPath))
+			namespace = GetPropertyValue(namespace, manifest.Package.Namespace, path.Base(manifestPath))
+			apiHost = GetPropertyValue(apiHost, manifest.Package.ApiHost, path.Base(manifestPath))
+		}
+	}
+
+	// Third, we need to look up the variables in .wskprops file.
+	pi := whisk.PropertiesImp{
+		OsPackage: whisk.OSPackageImp{},
+	}
+
+	// The error raised here can be neglected, because we will handle it in the end of this function.
+	wskprops, _ := GetWskPropFromWskprops(pi, proppath)
+	credential = GetPropertyValue(credential, wskprops.AuthKey, WSKPROPS)
+	namespace = GetPropertyValue(namespace, wskprops.Namespace, WSKPROPS)
+	apiHost = GetPropertyValue(apiHost, wskprops.APIHost, WSKPROPS)
+
+	// now, read credentials from whisk.properties but this is only acceptable within Travis
+	// whisk.properties will soon be deprecated and should not be used for any production deployment
+	whiskproperty, _ := GetWskPropFromWhiskProperty(pi)
+	credential = GetPropertyValue(credential, whiskproperty.AuthKey, WHISKPROPERTY)
+	if credential.Source == WHISKPROPERTY {
+		fmt.Println("WARNING: The authentication key was retrieved from whisk.properties " +
+			"which will soon be deprecated please do not use it outside of Travis builds.")
+	}
+	namespace = GetPropertyValue(namespace, whiskproperty.Namespace, WHISKPROPERTY)
+	if namespace.Source == WHISKPROPERTY {
+		fmt.Println("WARNING: The namespace was retrieved from whisk.properties " +
+			"which will soon be deprecated please do not use it outside of Travis builds.")
+	}
+	apiHost = GetPropertyValue(apiHost, whiskproperty.APIHost, WHISKPROPERTY)
+	if apiHost.Source == WHISKPROPERTY {
+		fmt.Println("WARNING: The API host was retrieved from whisk.properties " +
+			"which will soon be deprecated please do not use it outside of Travis builds.")
+	}
+
+	// set namespace to default namespace if not yet found
+	if len(apiHost.Value) != 0 && len(credential.Value) != 0 && len(namespace.Value) == 0 {
+		namespace.Value = whisk.DEFAULT_NAMESPACE
+		namespace.Source = DEFAULTVALUE
+	}
+
+	// If we still can not find the values we need, check if it is interactive mode.
+	// If so, we prompt users for the input.
+	// The namespace is set to a default value at this point if not provided.
+	if len(apiHost.Value) == 0 && isInteractive == true {
+		host, err := promptForValue("\nPlease provide the hostname for OpenWhisk [default value is openwhisk.ng.bluemix.net]: ")
+		utils.Check(err)
+		if host == "" {
+			host = "openwhisk.ng.bluemix.net"
+		}
+		apiHost.Value = host
+		apiHost.Source = INTERINPUT
+	}
+
+	if len(credential.Value) == 0 && isInteractive == true {
+		cred, err := promptForValue("\nPlease provide an authentication token: ")
+		utils.Check(err)
```

Comment:
Remove utils.check(), because we pop up all the error to the command line, and get rid of this method.